frame/staking/src/pallet/mod.rs

```diff
@@ -93,7 +93,7 @@ pub mod pallet {
 		>;
 
 		/// Maximum number of nominations per nominator.
-		const MAX_NOMINATIONS: u32;
+		type MaxNominations: Get<u32>;
```

Comment:
In the context of staking I think this is fine, but generally for bounded storage items I think the bound should be a const in the config and then `Get` can be implemented for that like https://github.com/paritytech/substrate/blob/master/frame/vesting/src/lib.rs#L138-L143

This ensures the value cannot be changed without a runtime upgrade, thus reducing a foot gun for pallet users

Reply:
fair point and will be considered. There's some minor benefit in less code (no wrapper), and keeping everything a simple `Get<_>`, so that they are all compatible, offloading the responsibility of not shooting themselves in the foot to the user. 

But I also see the point in your suggestion and there is certainly a way to keep this a `const`, while being able to use it as `Get<_>` in other places.

Reply:
although, this will put us back in the loop of why we have `Get<_>` and didn't use `const` from the get go: in this PR I've added tests where you want to change that parameter on the fly, and you can't do that with actual `const`. 

Given this, my verdict about this would be to postpone it to outside of this PR, since it is a FRAME-wise problem that exists in many places. I believe there might be an issue about this in the Frame Vision project. If not, I think you should make one about this potential footgun.

Reply:
I didn't find a pre-existing issue, but I made this: https://github.com/paritytech/substrate/issues/10613 . Feel free to close though if you think its a duplicate